src/drivers/generic/dummy-dma.c

```diff
@@ -184,11 +184,11 @@ static size_t dummy_dma_compute_avail_data(struct dma_chan_pdata *pdata)
  * @bytes bytes of data. Will copy exactly this much data if possible, however
  * it will stop short if you try to copy more data than available.
  */
-static size_t dummy_dma_do_copies(struct dma_chan_pdata *pdata, int bytes)
+static ssize_t dummy_dma_do_copies(struct dma_chan_pdata *pdata, int bytes)
 {
 	size_t avail = dummy_dma_compute_avail_data(pdata);
-	size_t copied = 0;
-	size_t crt_copied;
+	ssize_t copied = 0;
+	ssize_t crt_copied;
```

Comment:
I think crt_copied won't actually ever go below 0, so it could remain unsigned. It will only be done += copied when copied is >= 0. Return type should indeed be signed though, thanks.

Reply:
@paulstelian97 is crt_copied won't go below 0. I wanted to make code more consistent.th

Reply:
Fair enough then.